src/libs/actions/Report.js

```diff
@@ -1635,6 +1635,79 @@ function leaveRoom(reportID) {
     navigateToConciergeChat();
 }
 
+/**
+ * Flag a comment as offensive
+ *
+ * @param {String} reportID
+ * @param {Object} reportAction
+ * @param {String} severity
+ */
+function flagComment(reportID, reportAction, severity) {
+    const newDecision = {
+        "decision": "pending",
+    };
+
+    const message = reportAction.message[0];
+    const reportActionID = reportAction.reportActionID;
+
+    const updatedDecisions = [...(message.moderationDecisions || []), newDecision];
+
+    const updatedMessage = {
+        ...message,
+        moderationDecisions: updatedDecisions,
+    };
+
+    const optimisticReportActions = {
+        [reportActionID]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+            message: [updatedMessage],
+        },
+    };
```

Comment:
Any reason we're doing it like this instead of just adding it in value like for failure and success data?

Reply:
nope. probably just whatever I was using as a template.